github_cog.py

```
@@ -0,0 +1,111 @@
+import discord
+import github.GithubException
+from discord.ext import commands
+from github import Github
+from contextlib import closing
+import studybot
+
+
+# noinspection PyBroadException
+class GitHub_Integration(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name='github-join', help=studybot.help_dict.get('github-join'))
+    async def join(self, ctx, github_name='null'):
+        # Make sure they included the github_name arg, and send a response.
+        if github_name == 'null':
+            msg = 'You need to include your GitHub Username like this: '
+            msg += '`!github-join StudyBot`'
+            await ctx.message.reply(msg)
+            return
+        else:
+            g = Github(studybot.TOKENS.get('GITHUB_API_KEY'))
+            # Get the user object.
+            try:
+                g_user = g.get_user(github_name)
+            except:
+                await ctx.message.reply('The GitHub user you requested cannot be found. Double-Check your spelling.')
```

For all of the errors is there any logging or just replying? What if there is an issue with the reply? Would this error be lost?

---

There is logging but I didn't write it and it's completely undocumented, so I've got no idea how to use it.

---

I looked at the logger, it isn't quite correct. I will open a PR to fix it. But just ignore logging for now.